mantidproject/mantid
# Skip Sum Banks Algo for Linear detector

**Description of work.**
This PR ensures the Sum Banks algo is skipped if the loaded workspace has a linear detector. Otherwise, it will run the sum banks algo as usual.

**To test:**
1. Open the ISIS Reflectometry interface
2. Go to the Preview tab
3. Load `INTER45455_inst` with angle 1.0
4. There should be a plot in the Instrument View, Region Selection Viewer, and reduction plot
5. Load `INTER43538` (has linear detector)
6. The instrument view should be cleared, and a new plot appear in the Region Selection Viewer. (but note the reduction plot is not plotted).
7. Select a region which doesn't touch the edges of the plot. You should see a reduction plot

Fixes #34270

*This does not require release notes* because **the Refl Preview tab is still in development**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
